selenium_tests/base_page.py

```diff
@@ -0,0 +1,127 @@
+import logging
+from functools import partial
+
+from selenium.common.exceptions import (
+    NoSuchElementException,
+    InvalidSwitchToTargetException
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class BasePageMeta(type):
+    """
+    This metaclass allows us to avoid writing boilerplate that just finds
+    an element in a page by a locator tuple.
+
+    Expected usage:
+
+        class SomePageLocators(object):
+            TITLE = (By.CSS_SELECTOR, '#title')
+            BUTTON = (By.CSS_SELECTOR, '#the_only_button')
+
+        class SomePage(BasePage):
+            locator_class = SomePageLocators
+            located_properties = ['title', 'button']
```

Comment:
Any reason this isn't `locator_properties`? To me it reads like 'the list of properties that we located' vs. 'the list of locators that can be accessed as a class property'.